jukeInTheBoxUti/www/src/Bibliotheque.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div>
+    <h1>Juke in the box</h1>
+    <input
+      v-model="search"
+      v-on:input="getCatalogue()"
+      type="text"
+      class="barre"
+      id="search"
+      placeholder="rechercher"
+    >
+    <div class="bibliotheque">
+      <div id="wait"></div>
+      <div v-cloak class="biblio_pistes" v-for="piste in listMusiques">
+        <div class="piste">
+          <img class="img_piste" :src="piste.imagePiste">
+          <p>
+            <span v-for="artiste in piste.artistes">{{artiste.prénom}} {{artiste.nom}}</span>
+            - {{piste.nomPiste}}
+          </p>
+          <button class="add_button" v-on:click="addFile(piste.idPiste)">
+            Ajouter à la
+            file
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  data() {
+    return { listMusiques: "", search: "" };
+  },
+  methods: {
+    //retourne le catalogue du jukebox
+    getCatalogue: function() {
+        //si le token est valide alors on effectue la requete
+      //sinon on redemande un token valide (cf tokenValide)
+      if (this.$parent.tokenValide()) {
+        axios
+          .get(this.$parent.url + "catalogue", {
+            params: {
+              piste: this.search,
+              token: this.$parent.token
+            }
+          })
+          .then(response => {
+            this.listMusiques = response["data"]["catalogue"]["pistes"];
+          });
+      } else {
+        listMusiques = "";
+      }
+    },
+     //ajoute une musique du catalogue dans la file du jukebox
+    addFile: function(idPiste) {
+        //si le token est valide alors on effectue la requete
+      //sinon on redemande un token valide (cf tokenValide)
+      if (this.$parent.tokenValide()) {
```

Comment:
C'est un antipatern que d'utiliser this.$parent. Si vous avez besoin d'un paramètre ou d'une méthode du parent, ils doivent venir via des props.

Je vous conseiller de regarder du côté de [vuex](https://vuex.vuejs.org).